boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,156 @@
+
+    private static final Logger LOGGER = Logger.getLogger(IAMStarter.class);
+
+    private static final String SEPARATOR = File.separator;
+
+    private static final String MICROSERVICE_NAME = "iam.war";
+
+    private static final String READY_MARKER = "Ready to process requests";
+
+    private static final int TIMEOUT = 180;
+
+    private static Process process;
+
+    private static List<String> command = new ArrayList<>();
+
+    private static boolean started = false;
+
+    /**
+     *  Start IAM microservice
+     */
+    public static Process start(String paHome, String microservicesPath, String[] jvmArgs)
+            throws InterruptedException, IOException, ExecutionException {
+
+        if (!started && buildJavaCommand(paHome, jvmArgs) && buildMicroservicePath(paHome, microservicesPath)) {
+
+            LOGGER.info("Starting IAM microservice");
+
+            ProcessBuilder processBuilder = new ProcessBuilder(command);
+            processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+
+            process = processBuilder.start();
+            LOGGER.info(streamOutput(process.getInputStream()));
+
+            started = true;
+        }
+        return process;
+    }
+
+    /**
+     * Stream microservice output (using an executor) to check that it starts properly
+     */
+    private static String streamOutput(InputStream inputStream)
+            throws IOException, InterruptedException, ExecutionException {
+
+        // Stream microservice output
+        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<String> future = executor.submit(new Callable() {
+
+            public String call() throws IOException {
+
+                try {
+                    String line = null;
+
+                    while ((line = br.readLine()) != null) {
+                        System.out.print(".");
+                        if (line.contains(READY_MARKER)) {
+                            break;
+                        }
+                    }
+                } finally {
+                    br.close();
+                }
+                return "\nIAM Microservice started";
+            }
+        });
+
+        // Check for timeout
+        try {
+            return future.get(TIMEOUT, TimeUnit.SECONDS);
+        } catch (TimeoutException toe) {
+            // Stop streaming the output, but the microservice continues to execute
+            br.close();
+            LOGGER.warn("Warning: IAM Microservice timed out to start. See the logs for the details.");
+            return "\nWarning: IAM Microservice timed out to start. See the logs for the details.";
+        } finally {
+            executor.shutdownNow();
+            future.cancel(true);
+        }
+    }
+
+    /**
+     * Prepare java command with jvm args
+     */
+    private static boolean buildJavaCommand(String pa_home, String[] jvmArgs) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)